### PR TITLE
waf: cross compilation enhancements

### PR DESCRIPTION
**Pull request by guludo**
Hi all,

This is an enhancement alternative for coping with different configuration methods for gcc and clang.

@OXINARF, this is a followup of https://github.com/diydrones/ardupilot/pull/3765#discussion_r56691973.

Best,
Gustavo Sousa

### CODE REVIEW

File: Tools/ardupilotwaf/toolchain.py [hunk 1]
```diff
@@ -1,93 +1,120 @@
...
-                else:
-                    return os.path.dirname(realpath)
+def configure(cfg):
+    if cfg.env.TOOLCHAIN == 'native':
+        cfg.load('compiler_cxx compiler_c')
+        return
 
-            else:
-                return os.path.dirname(path)
+    cfg.env.AR = cfg.env.TOOLCHAIN + '-ar'
```
Comment (OXINARF) [inline Tools/ardupilotwaf/toolchain.py]: This is forcing AR to always be in the format _prefix_-ar, it should only be set for GCC or Clang.

**Comment by OXINARF** [comment]
@guludo I like it. It seems to be everything OK, except the comment I made.

File: Tools/ardupilotwaf/toolchain.py (hunk 1 above)
Comment (guludo) [inline Tools/ardupilotwaf/toolchain.py]: I thought about that, but right now those are the only ones we support for cross compilation. Maybe an alternative is skipping compilers that are not supported for cross-compilation.

**Comment by guludo** [comment]
@OXINARF 
I just added a new commit forcing cross-compilation support only for gnu and clang compilers.

**Comment by lucasdemarchi** [comment]
Applied, thanks

**Pull request closed without merging**